clientconn.go

```diff
@@ -71,6 +71,18 @@ type dialOptions struct {
 // DialOption configures how we set up the connection.
 type DialOption func(*dialOptions)
 
+// NameResolver defines the functions needed to resolve a customized names.
+type NameResolver interface {
+	// Get returns the map of customized names and the corresponding ip/DNS addresses.
+	Get(target string) map[string]string
+	// Watch set a watcher on the target directory/key.
+	Watch(target string)
```

Comment:
Watch watches for the name resolution changes on target. It blocks until Stop() is invoked. The watch results are obtained via GetUpdate().